src/AutoMapper/Mapper.cs

```diff
@@ -84,7 +84,7 @@ public interface IMapper : IMapperBase
     /// <param name="destinationType">Destination type to create</param>
     /// <param name="opts">Mapping options</param>
     /// <returns>Mapped destination object</returns>
-    object Map(object source, Type sourceType, Type destinationType, Action<IObjectMappingOperationOptions> opts);
+    object? Map(object? source, Type? sourceType, Type destinationType, Action<IObjectMappingOperationOptions> opts);
```

Comment:
Technically `destinationType` can be null here, it would fallback to object. But null doesn't really makes sense, so I want to say this overload is different.